lib/netsparker/vulnerability.rb

```diff
@@ -105,19 +105,35 @@ def cleanup_html(source)
       result.gsub!(/&lt;/, '<')
       result.gsub!(/&gt;/, '>')
 
-      result.gsub!(/<b>(.*?)<\/b>/) { "*#{$1.strip}*" }
-      result.gsub!(/<br\/>/, "\n")
-      result.gsub!(/<code><pre.*?>(.*?)<\/pre><\/code>/m){|m| "\n\nbc.. #{$1.strip}\n\np.  \n" }
-      result.gsub!(/<div>(.*?)<\/div>/, '\1 ')
-      result.gsub!(/<em>(.*?)<\/em>/, '\1')
+      result.gsub!(/<b>(.*?)<\/b>/) { "*#{$1}*" }
+      result.gsub!(/<i>(.*?)<\/i>/) { "_#{$1}_" }
+      result.gsub!(/<em>(.*?)<\/em>/) { "*#{$1}*" }
+      result.gsub!(/<h2>(.*?)<\/h2>/) { "*#{$1}*" }
+      result.gsub!(/<strong>(.*?)<\/strong>/) { "*#{$1}*" }
+
+      result.gsub!(/(<br>)|(<br\/>)/, "\n")
+
+      result.gsub!(/<div>(.*?)<\/div>/m, '\1')
       result.gsub!(/<font.*?>(.*?)<\/font>/m, '\1')
-      result.gsub!(/<h2>(.*?)<\/h2>/) { "*#{$1.strip}*" }
-      result.gsub!(/<i>(.*?)<\/i>/, '\1')
-      result.gsub!(/<li.*>(.*)?<\/li>/, '* \1')
-      result.gsub!(/<p>(.*?)<\/p>/, '\1 ')
-      result.gsub!(/<pre.*?>(.*?)<\/pre>/m){|m| "\n\nbc.. #{$1.strip}\n\np.  \n" }
-      result.gsub!(/<span.*>(.*?)<\/span>/, '\1')
-      result.gsub!(/<ul>(.*?)<\/ul>/m, '\1')
+      result.gsub!(/<p (.*?)>(.*?)<\/p>/) { "\n#{$2}\n" }
+      result.gsub!(/<span(.*?)>(.*?)<\/span>/, '\2')
+      result.gsub!(/(<p>)|(<\/p>)/, "\n")
+
+      result.gsub!(/<a href=\"(.*?)\" (.*?)>(.*?)<\/a>/i) { "\"#{$3.strip}\":#{$1.strip}" }
+
+      result.gsub!(/<code><pre.*?>(.*?)<\/pre><\/code>/) {"\n\nbc. #{$1}\n\n" }
```

Comment:
may be we can merge the `gsub('<code><pre>')` calls?
Or maybe merge all `gsub` calls that transform xml with `code` and `pre` could be handled by a single `gsub` call?